Hearthstone Deck Tracker/OpponentDeadForTracker.cs

```diff
@@ -0,0 +1,60 @@
+﻿using HearthDb.Enums;
+using Hearthstone_Deck_Tracker.Hearthstone;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Hearthstone_Deck_Tracker
+{
+	public static class OpponentDeadForTracker
+	{
+		private static List<string> _uniqueDeadHeroes = new List<string>();
+		private static List<int> _deadTracker = new List<int>();
+		const string KelThuzadCardId = "KelThuzad";
+
+		public static async void ShoppingStarted(GameV2 game)
+		{
+			if(game.GetTurnNumber() <= 1)
+			{
+				_uniqueDeadHeroes.Clear();
+				_deadTracker.Clear();
+				Core.Overlay.nextOpponentLeaderboardPosition = null;
+			}
+			for(int i =0; i < _deadTracker.Count; i++)
+			{
+				_deadTracker[i]++;
+			}
+			var deadHeroes = game.Entities.Select(x=>x.Value).Where(x => x.IsHero && x.Health <= 0);
+			foreach(var hero in deadHeroes)
+			{
+				var id = BattlegroundsBoardState.GetCorrectBoardstateHeroId(hero.CardId);
+				if(!id.Contains(KelThuzadCardId) && !_uniqueDeadHeroes.Contains(id))
+				{
+					_deadTracker.Add(0);
+					_uniqueDeadHeroes.Add(id);
+				}
+			}
+			_deadTracker.Sort((x, y) => y.CompareTo(x));
+			Core.Overlay.UpdateOpponentDeadForTurns(_deadTracker);
+			var gameEntites = game.Entities.Values;
+			var currentPlayer = gameEntites.FirstOrDefault(x => x.IsCurrentPlayer);
+			for(int i = 0; i < 5; i++)
+			{
+				if(currentPlayer != null && currentPlayer.HasTag(GameTag.NEXT_OPPONENT_PLAYER_ID))
+				{
+					var nextOpponent = gameEntites.FirstOrDefault(x => x.GetTag(GameTag.PLAYER_ID) == currentPlayer.GetTag(GameTag.NEXT_OPPONENT_PLAYER_ID));
+					if(nextOpponent != null)
+					{
+						var leaderboardPlace = nextOpponent.GetTag(GameTag.PLAYER_LEADERBOARD_PLACE);
+						if(leaderboardPlace >= 0 && leaderboardPlace < 8)
+						{
+							Core.Overlay.nextOpponentLeaderboardPosition = leaderboardPlace;
+							Core.Overlay.PositionDeadForText();
```

Comment:
Instead of updating a field on another class, that is used for a subsequent call, calling `Core.Overlay.PositionDeadForText(leaderboardPlace)` would be a lot cleaner.